pkg/starkapi/forms_api.go

```diff
@@ -80,30 +89,67 @@ func (formsApi *FormsApi) GetControlByName(name string) (domain.FormControl, err
 	return control, nil
 }
 
-func (formsApi *FormsApi) CreateControlOnRef(control domain.FormControl) (domain.FormControl, error) {
-	err := control.Validate()
+func (formsApi *FormsApi) CreateControlOnRef(formControlName string, ref string, value string) (domain.FormControlRef, error) {
+	var control domain.FormControl
+	var controlRef domain.FormControlRef
+
+	err := controlRef.ValidateStringParams(formControlName, errNoControlNameProvided)
 	if err != nil {
-		return control, err
+		logger.Error(err)
+		return controlRef, err
+	}
+
+	err = controlRef.ValidateStringParams(ref, errNoRefProvided)
+	if err != nil {
+		logger.Error(err)
+		return controlRef, err
 	}
 
-	url := fmt.Sprintf("%s/%s%s", formsApi.baseUrl(), control.Ref, formsApi.controlsPrefix())
+	err = controlRef.ValidateStringParams(value, errNoValueProvided)
+	if err != nil {
+		logger.Error(err)
+		return controlRef, err
+	}
 
-	body, err := json.Marshal(control)
+	control, err = formsApi.GetControlByName(formControlName)
 	if err != nil {
-		return control, err
+		logger.Error(err)
+		return controlRef, err
+	}
+
+	if control.Id < 1 || control.Name != formControlName {
+		newError := errors.New(fmt.Sprintf(errInvalidControlNameProvided, formControlName))
+		logger.Error(newError)
```

Comment:
please add a meaningful description to the logger.Error

Reply:
I believe this already has a meaningful error. "no form control found with name provided : [%s], formControlName". If this requires more information then re-request changes please. Thanks